File: mobile/SmartSolarMobile/app/src/main/java/com/smartsolar/mobile/data/remote/dto/ReservationVerificationResponse.java
package com.smartsolar.mobile.data.remote.dto;

public class ReservationVerificationResponse {
    private String reservationId;
    private String prosumerNic;
    private String stationId;
    private String slotId;
    private double energyAmountKwh;
    private String scheduledStartAtUtc;
    private String scheduledEndAtUtc;
    private String status;
    private String qrIssuedAtUtc;
    private boolean eligibleForCompletion;

    public ReservationVerificationResponse() { }

    public ReservationVerificationResponse(String reservationId, String prosumerNic, String stationId,
                                           String slotId, double energyAmountKwh, String scheduledStartAtUtc,
                                           String scheduledEndAtUtc, String status, String qrIssuedAtUtc,
                                           boolean eligibleForCompletion) {
        this.reservationId = reservationId;
        this.prosumerNic = prosumerNic;
        this.stationId = stationId;
        this.slotId = slotId;
        this.energyAmountKwh = energyAmountKwh;
        this.scheduledStartAtUtc = scheduledStartAtUtc;
        this.scheduledEndAtUtc = scheduledEndAtUtc;
        this.status = status;
        this.qrIssuedAtUtc = qrIssuedAtUtc;
        this.eligibleForCompletion = eligibleForCompletion;
    }

    public String getReservationId() { return reservationId; }
    public String getProsumerNic() { return prosumerNic; }
    public String getStationId() { return stationId; }
    public String getSlotId() { return slotId; }
    public double getEnergyAmountKwh() { return energyAmountKwh; }
    public String getScheduledStartAtUtc() { return scheduledStartAtUtc; }
    public String getScheduledEndAtUtc() { return scheduledEndAtUtc; }
    public String getStatus() { return status; }
    public String getQrIssuedAtUtc() { return qrIssuedAtUtc; }
    public boolean isEligibleForCompletion() { return eligibleForCompletion; }
}

File: src/SmartSolar.Application/DTOs/Reservations/ReservationVerificationResponse.cs
/*
 * File: ReservationVerificationResponse.cs
 * Project: Smart Solar Microgrid Trading System
 * Purpose: Returns trusted authoritative reservation details after server QR verification.
 * Note: Keep this header and update method-level comments as the code evolves.
 */

using SmartSolar.Domain.Enums;

namespace SmartSolar.Application.DTOs.Reservations;

public sealed record ReservationVerificationResponse(
    string ReservationId,
    string ProsumerNic,
    string StationId,
    string SlotId,
    decimal EnergyAmountKwh,
    DateTime ScheduledStartAtUtc,
    DateTime ScheduledEndAtUtc,
    ReservationStatus Status,
    DateTime? QrIssuedAtUtc,
    bool EligibleForCompletion
);
